File: split/eqsave.pas
procedure eqsave(p: halfword; l: quarterword);
begin
  if saveptr > maxsavestack then
  begin
    maxsavestack := saveptr;
    if maxsavestack > savesize - 6 then
      overflow(541, savesize);
  end;
  if l = 0 then
    savestack[saveptr].hh.b0 := 1
  else
  begin
    savestack[saveptr] :=
      eqtb[p];
    saveptr := saveptr + 1;
    savestack[saveptr].hh.b0 := 0;
  end;
  savestack[saveptr].hh.b1 := l;
  savestack[saveptr].hh.rh := p;
  saveptr := saveptr + 1;
end;

File: cpp/src/eqsave.cpp
#include "eqsave.h"
#include "erreur.h"

void eqsave(halfword p, quarterword l)
{
	check_full_save_stack(curalign);
	if (l == level_zero)
		save_type(saveptr) = restore_zero;
	else
	{
		saved(0) = eqtb[p].int_;
		saveptr++;
		save_type(saveptr) = restore_old_value;
	}
	save_level(saveptr) = l;
	save_index(saveptr) = p;
	saveptr++;
}
